package/samplers/hebo/sampler.py

```diff
@@ -103,3 +230,26 @@ def _convert_to_hebo_design_space(
             else:
                 raise NotImplementedError(f"Unsupported distribution: {distribution}")
         return DesignSpace().parse(design_space)
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        return optuna.search_space.intersection_search_space(
+            study._get_trials(deepcopy=False, use_cache=True)
+        )
+
+    def sample_independent(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        if not self._is_fallback_inevitable:
```

Comment:
Probably, warning happens when fallback is inevitable, but when it is NOT inevitable?